pilot/pkg/bootstrap/server.go

```diff
@@ -455,13 +451,14 @@ func (s *Server) initMCPConfigController(args *PilotArgs) error {
 			if err != nil {
 				return err
 			}
-			credentials := creds.CreateForClient(u.Hostname(), watcher)
+			credentials := creds.CreateForClient(configSource.TlsSettings.Sni, watcher)
```

Comment:
what happens if SNI is emtpy?

Reply:
if SNI is empty, tls.Config.ServerName is set with `configSource.Address` here.

crypto/tls/tls.go 
```
	// If no ServerName is set, infer the ServerName
	// from the hostname we're connecting to.
	if config.ServerName == "" {
		// Make a copy to avoid polluting argument or default.
		c := config.Clone()
		c.ServerName = hostname
		config = c
	}

```